receiver/awscontainerinsightreceiver/internal/stores/podresourcesstore_test.go

```diff
@@ -0,0 +1,163 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package stores // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/stores"
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/zap"
+	podresourcesv1 "k8s.io/kubelet/pkg/apis/podresources/v1"
+)
+
+var (
+	expectedContainerInfoToResourcesMap = map[ContainerInfo][]ResourceInfo{
+		{
+			podName:       "test-pod",
+			containerName: "test-container",
+			namespace:     "test-namespace",
+		}: {
+			{
+				resourceName: "test-resource",
+				deviceID:     "device-id-1",
+			},
+			{
+				resourceName: "test-resource",
+				deviceID:     "device-id-2",
+			},
+		},
+	}
+
+	expectedResourceToPodContainerMap = map[ResourceInfo]ContainerInfo{
+		{
+			resourceName: "test-resource",
+			deviceID:     "device-id-1",
+		}: {
+			podName:       "test-pod",
+			containerName: "test-container",
+			namespace:     "test-namespace",
+		},
+		{
+			resourceName: "test-resource",
+			deviceID:     "device-id-2",
+		}: {
+			podName:       "test-pod",
+			containerName: "test-container",
+			namespace:     "test-namespace",
+		},
+	}
+
+	expectedContainerInfo = ContainerInfo{
+		podName:       "test-pod",
+		containerName: "test-container",
+		namespace:     "test-namespace",
+	}
+
+	expectedResourceInfo = []ResourceInfo{
+		{
+			resourceName: "test-resource",
+			deviceID:     "device-id-1",
+		},
+		{
+			resourceName: "test-resource",
+			deviceID:     "device-id-2",
+		},
+	}
+)
+
+type MockPodResourcesClient struct {
+}
+
+func (m *MockPodResourcesClient) ListPods() (*podresourcesv1.ListPodResourcesResponse, error) {
+	mockResp := &podresourcesv1.ListPodResourcesResponse{
+		PodResources: []*podresourcesv1.PodResources{
+			{
+				Name:      "test-pod",
+				Namespace: "test-namespace",
+				Containers: []*podresourcesv1.ContainerResources{
+					{
+						Name: "test-container",
```

Comment:
Can you also add a container with no devices to test that nothing is stored for that one.

And also a test case with a completely empty response to make sure the code doesn't break.